codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2RemoteFlowController.java

```diff
@@ -57,15 +58,30 @@ public DefaultHttp2RemoteFlowController(Http2Connection connection) {
         connection.addListener(new Http2ConnectionAdapter() {
             @Override
             public void onStreamAdded(Http2Stream stream) {
-                // Just add a new flow state to the stream.
-                stream.remoteFlowState(new DefaultFlowState(stream, 0));
+                // If the stream state is not open then the stream is not yet eligible for flow controlled frames and
+                // only requires the ReducedFlowState. Otherwise the full amount of memory is required.
+                stream.remoteFlowState(stream.state() == IDLE ?
```

Comment:
> This is no longer the case since this PR was opened.

Why not? It's reasonable to expect that any stream in idle will eventually go to open. So you will always do more allocations. 

> but based on the WG discussion it sounds like the IDLE streams are intended to be kept in IDLE for longer periods of time.

With "longer periods of time" being milliseconds to a few seconds. 

I see no point in this PR.